### PR TITLE
Remove the setter from IAnalysisIssueSelectionService

### PR DESCRIPTION
**Pull request by rita-g-sonarsource**
Part of #2023

### CODE REVIEW

File: src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs [hunk 1]
```diff
@@ -34,67 +34,90 @@ namespace SonarLint.VisualStudio.IssueVisualization.UnitTests
...
+        {
+            MefTestHelpers.CheckTypeCanBeImported<AnalysisIssueSelectionService, IAnalysisIssueSelectionService>(null,
+                new[]
+                {
+                    MefTestHelpers.CreateExport<SVsServiceProvider>(Mock.Of<IServiceProvider>()),
+                    MefTestHelpers.CreateExport<IIssueSelectionService>(Mock.Of<IIssueSelectionService>())
+                });
+        }
 
-        [TestInitialize]
```
Comment (rita-g-sonarsource) [inline src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs:41]: I replaced the TestInitialize with a CreateTestSubject method.
The tests were updated accordingly.

File: src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs [hunk 2]
```diff
@@ -34,67 +34,90 @@ namespace SonarLint.VisualStudio.IssueVisualization.UnitTests
...
-        public void TestInitialize()
+        [TestMethod]
+        public void Ctor_RegisterToIssueSelectionEvent()
```
Comment (rita-g-sonarsource) [inline src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs:49]: New test

File: src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs [hunk 3]
```diff
@@ -34,67 +34,90 @@ namespace SonarLint.VisualStudio.IssueVisualization.UnitTests
...
+            CreateTestSubject(selectionService: selectionService.Object);
 
-            testSubject = new AnalysisIssueSelectionService(serviceProviderMock.Object);
+            selectionService.VerifyAdd(x => x.SelectedIssueChanged += It.IsAny<EventHandler>(), Times.Once);
+            selectionService.VerifyNoOtherCalls();
         }
 
         [TestMethod]
-        public void MefCtor_CheckIsExported()
+        public void Dispose_UnregisterFromIssueSelectionEvent()
```
Comment (rita-g-sonarsource) [inline src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs:61]: New test

File: src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs [hunk 4]
```diff
@@ -274,41 +322,83 @@ public void SelectedIssueChanged_ChangesSelectedFlow()
...
+            var secondIssue = CreateIssue(secondIssueFirstFlow);
             testSubject.SelectedIssue = secondIssue;
             eventHandler.Verify(x => x(testSubject, It.Is((SelectionChangedEventArgs args) => args.SelectedFlow == secondIssueFirstFlow)), Times.Once());
             testSubject.SelectedFlow.Should().Be(secondIssueFirstFlow);
 
             eventHandler.VerifyNoOtherCalls();
         }
 
-        private static IAnalysisIssueFlowVisualization GetFlowWithLocation(out IAnalysisIssueLocationVisualization firstLocation)
+        [TestMethod]
```
Comment (rita-g-sonarsource) [inline src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs:334]: 3 new tests that used to be in IssueSelectionServiceTests: null issue, issue without SLs, and issue with SLs.

File: src/IssueViz/Selection/IAnalysisIssueSelectionService.cs [hunk 5]
```diff
@@ -27,7 +27,7 @@ public interface IAnalysisIssueSelectionService : IDisposable
     {
         event EventHandler<SelectionChangedEventArgs> SelectionChanged; 
 
-        IAnalysisIssueVisualization SelectedIssue { get; set; }
+        IAnalysisIssueVisualization SelectedIssue { get; }
```
Comment (rita-g-sonarsource) [inline src/IssueViz/Selection/IAnalysisIssueSelectionService.cs:30]: 🎉 !

File: src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs [hunk 6]
```diff
@@ -274,41 +322,83 @@ public void SelectedIssueChanged_ChangesSelectedFlow()
...
-            mockIssue
-                .Setup(x => x.Flows)
-                .Returns(new[] { firstFlow, otherFlow });
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
 
-            return mockIssue.Object;
+            RaiseSelectedIssueChangedEvent(selectionService, issue);
+
+            testSubject.SelectedIssue.Should().BeNull();
```
Comment (duncanp-sonar) [inline src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs:361]: Was `testSubject.SelectedIssue` ever set to anything other than `null`?
If not, this test doesn't test that the issue is cleared. Same for the next test.

**Review by duncanp-sonar — COMMENTED** [review]
Looks good, just a question about two of the new tests.

**Pull request merged**